src/kubectlUtils.ts

```diff
@@ -158,15 +158,54 @@ export async function deleteCluster(kubectl: Kubectl, context: KubectlContext):
     return true;
 }
 
+export async function getAsDataResource(resourceId: string, kind: string, kubectl: Kubectl): Promise<DataResource> {
+    const currentNS = await currentNamespace(kubectl);
+
+    const dataResource = await kubectl.asJson<DataResource>(`get ${kind} ${resourceId} -o json --namespace=${currentNS}`);
+    if (failed(dataResource)) {
+        vscode.window.showErrorMessage(dataResource.error[0]);
+        return {
+            data: {},
+            kind,
+            metadata: {
+                name: resourceId,
+                namespace: currentNS
+            }
+        };
+    }
+    return dataResource.result;
+}
+
 export async function getAsDataResources(resource: string, kubectl: Kubectl): Promise<DataResource[]> {
     const currentNS = await currentNamespace(kubectl);
 
-    const resources = await kubectl.asJson<KubernetesCollection<DataResource>>(`get ${resource} -o json --namespace=${currentNS}`);
+    const resources = await kubectl.asLines(`get ${resource} --show-labels --namespace=${currentNS}`);

```

Comment:
Should we explicitly set the `-o` flag here? IIRC kubernetes does not guarantee that the `--output` flag's default value will remain consistent, and could be subject to change.

Reply:
Also would the `--output jsonpath` be a more portable/stable option than parsing the output from `--show-labels`?

Reply:
AFAIK jsonpath is not faster than `--show-labels` and it doesn't solve our problem. I didn't dig up how `get resource --show-labels` works but i guess that this command gets the data directly while jsonpath parses all content trying to find a match with what's in the jsonpath expression. And it takes time. But i have to admit that i'm not an expert and maybe there are ways to write jsonpath expressions which perform better than those i write.

Reply:
For example, this is the result by trying to only take the secrets names with jsonpath.
```
kubectl get secrets -o jsonpath="{range .items[*]}{.metadata.name}{end}"

real	0m21.091s
user	0m16.469s
sys	0m0.538s
```